relay-server/src/metrics_extraction/transactions/mod.rs

```diff
@@ -375,6 +380,76 @@ fn extract_transaction_metrics_inner(
     Ok(())
 }
 
+fn extract_span_metrics(
+    aggregator_config: &AggregatorConfig,
+    event: &mut Event,
+    metrics: &mut Vec<Metric>, // output parameter
+) -> Result<(), ExtractMetricsError> {
+    // TODO(iker): measure the performance of this whole method
+
+    if event.ty.value() != Some(&EventType::Transaction) {
+        return Ok(());
+    }
+    let (Some(_), Some(&end)) = (event.start_timestamp.value(), event.timestamp.value()) else {
+        relay_log::debug!("failed to extract the start and the end timestamps from the event");
+        return Err(ExtractMetricsError::MissingTimestamp);
+    };
+
+    let Some(timestamp) = UnixTimestamp::from_datetime(end.into_inner()) else {
+        relay_log::debug!("event timestamp is not a valid unix timestamp");
+        return Err(ExtractMetricsError::InvalidTimestamp);
+    };
+
+    // Validate the transaction event against the metrics timestamp limits. If the metric is too
+    // old or too new, we cannot extract the metric and also need to drop the transaction event
+    // for consistency between metrics and events.
+    if !aggregator_config.timestamp_range().contains(&timestamp) {
+        relay_log::debug!("event timestamp is out of the valid range for metrics");
+        return Err(ExtractMetricsError::InvalidTimestamp);
+    }
+
```

Comment:
This part (all the validation thingy) could be refactored in the separate helper function, since it's already used in the `extract_transaction_metrics_inner` in the same way

Reply:
Yes, I agree. I think there's some refactoring that must happen from this PR.